### ISSUE TITLE
feature_request(anchors): capital letters in headers support

### ISSUE THREAD

**Issue by Kristinita** [remarkjs/remark-validate-links#32]
### 1. Summary

**If** capital letters an anchor names:

    it would be nice, if remark-validate-links will not mark it as warnings.

### 2. Argumentation

Capital letters in URLs is valid: URLs will be open i browsers.

I use capital letters for more readable URLs; see [**answer**](https://stackoverflow.com/a/33489532/5951529) for details.

### 3. Data

+ `Kira_remark-valdate-links__capital_letters_in_headers--non-valid_demo.md`

```md
<!-- MarkdownTOC -->

1. [Kira](#Kira)
1. [isGoddess](#isGoddess)
1. [Кира](#Кира)
1. [этоБогиня](#этоБогиня)

<!-- /MarkdownTOC -->

<a id="Kira"></a>
# Kira

<a id="isGoddess"></a>
# isGoddess

<a id="Кира"></a>
# Кира

<a id="этоБогиня"></a>
# этоБогиня

```

### 4. Expected behavior

No warnings.

### 5. Actual behavior

```text
D:\SashaDebugging>remark Kira_remark-valdate-links__capital_letters_in_headers--non-valid_demo.md --use remark-validate-links --no-stdout
Kira_remark-valdate-links__capital_letters_in_headers--non-valid_demo.md
  3:4-3:17  warning  Link to unknown heading: `Kira`. Did you mean `kira`            missing-heading                  remark-validate-links
  4:4-4:27  warning  Link to unknown heading: `isGoddess`. Did you mean `isgoddess`  missing-heading                  remark-validate-links
  5:4-5:17  warning  Link to unknown heading: `Кира`. Did you mean `кира`            missing-heading                  remark-validate-links
  6:4-6:27  warning  Link to unknown heading: `этоБогиня`. Did you mean `этобогиня`  missing-heading                  remark-validate-links

‼ 5 warnings
```

Thanks.

**Comment by ChristianMurphy** [comment]
Ids are processed with GitHub Slugger, which lowercases content by default
https://github.com/remarkjs/remark-validate-links/blob/872b653b57dfa785fe6d84b05291f6d1f8668132/index.js#L189-L191
Slugger has a `maintainCase` option which may resolve the issue you are seeing. https://github.com/Flet/github-slugger/blob/master/index.js#L17-L23
Try out adding `maintainCase` option support to `remark-validate-links`, and if it works feel free to open a PR!

**Comment by wooorm** [comment]
The problem here is slightly different I think. remark-validate-links expects GitHub-style slugging. But your example has hardcoded HTML references.

Say you had:

```markdown
<a id="Foo"></a>
# Kira
```

...remark-validate-links doesn’t pick up on the `id`s: it only sees `kira` from the heading.

**Comment by wooorm** [comment]
Related to https://github.com/remarkjs/remark-validate-links/issues/1

**Comment by wooorm** [comment]
I’m closing this as full-on HTML parsing is not in the scope of this project. However, as mentioned in GH-1 you can integrate with this plugin by parsing the HTML yourself, looking for `id`s and `name`s, and exposing them on `node.data.id` (where `node` is the HTML node).

[This plugin then picks up them](https://github.com/remarkjs/remark-validate-links#integration)!
